server/sonar-db-dao/src/main/java/org/sonar/db/ce/CeQueueDao.java

```diff
@@ -84,6 +85,10 @@ public int countByQuery(DbSession dbSession, CeTaskQuery query) {
     return mapper(dbSession).selectPendingByMinimumExecutionCount(minExecutionCount);
   }
 
+  public void resetTasksWithUnknownWorkerUUIDs(DbSession dbSession, Set<String> knownWorkerUUIDs) {
+    mapper(dbSession).resetTasksWithUnknownWorkerUUIDs(knownWorkerUUIDs, system2.now());
```

Comment:
If my understanding is correct the executeLargeUpdates is used when parameters are huge in order to make partition of a Collection. In our case, in a datacenter edition we'll have only 3 servers with let say 2 worker for each server so at most it will be a Set of 6 string. We are far enough from the limit that it's 1000.
The method executeLargeUpdates is transforming the Set to a List so it will require to update resetTasksWithUnknownWorkerUUIDs to take a List, it's not a big deal but we are not expecting any duplication in this List so it's really a Set.
Is it really valuable to use executeLargeUpdates with a Set that will contains only 6 elements ?